Segmentation fault when creating new certificate

**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
I installed cert-manager and I tried to issue a certificate. `cert-manager` segfaulted

**What you expected to happen**:
I expected the certificate to be created

**How to reproduce it (as minimally and precisely as possible)**:
Apply the following yml file:
```
---
# Source: cert-manager/templates/issuer-crd.yaml

apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: issuers.certmanager.k8s.io
spec:
  group: certmanager.k8s.io
  version: v1alpha1
  names:
    kind: Issuer
    plural: issuers
  scope: Namespaced

---
# Source: cert-manager/templates/certificate-crd.yaml
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: certificates.certmanager.k8s.io
spec:
  group: certmanager.k8s.io
  version: v1alpha1
  names:
    kind: Certificate
    plural: certificates
  scope: Namespaced

---
# Source: cert-manager/templates/clusterissuer-crd.yaml

apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: clusterissuers.certmanager.k8s.io
spec:
  group: certmanager.k8s.io
  version: v1alpha1
  names:
    kind: ClusterIssuer
    plural: clusterissuers
  scope: Cluster

---
# Source: cert-manager/templates/deployment.yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: cert-manager
  labels:
    app: cert-manager
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: cert-manager
    spec:
      serviceAccount: cert-manager
      serviceAccountName: cert-manager
      containers:
        - name: cert-manager
          image: quay.io/jetstack/cert-manager-controller:v0.2.1
          imagePullPolicy: Always
---
apiVersion: certmanager.k8s.io/v1alpha1
kind: ClusterIssuer
metadata:
  name: letsencrypt-staging
spec:
  acme:
    server: https://acme-staging.api.letsencrypt.org/directory
    email: user@example.com
    privateKeySecretRef:
      name: letsencrypt-staging-issuer-account-key
```

Then create a new certificate using:

```
apiVersion: certmanager.k8s.io/v1alpha1
kind: Certificate
metadata:
  name: builder-tls
  namespace: ci
spec:
  secretName: builder-tls
  issuerRef:
    name: letsencrypt-staging
    kind: ClusterIssuer
  dnsNames:
    - builder.int.company.com
```


**Anything else we need to know?**:
Stacktrace:
```
I1128 10:00:33.025217       1 leaderelection.go:174] attempting to acquire leader lease...
I1128 10:00:33.025356       1 server.go:68] Listening on http://0.0.0.0:9402
I1128 10:00:51.159009       1 leaderelection.go:184] successfully acquired lease kube-system/cert-manager-controller
I1128 10:00:51.261609       1 controller.go:138] clusterissuers controller: syncing item 'letsencrypt-staging'
I1128 10:00:51.997304       1 helpers.go:122] Setting lastTransitionTime for ClusterIssuer "letsencrypt-staging" condition "Ready" to 2017-11-28 10:00:51.99728891 +0000 UTC
I1128 10:00:52.004898       1 controller.go:152] clusterissuers controller: Finished processing work item "letsencrypt-staging"
I1128 10:00:52.005001       1 controller.go:138] clusterissuers controller: syncing item 'letsencrypt-staging'
I1128 10:00:52.638406       1 controller.go:152] clusterissuers controller: Finished processing work item "letsencrypt-staging"
I1128 10:01:07.298559       1 controller.go:187] certificates controller: syncing item 'ci/builder-tls'
I1128 10:01:07.298678       1 sync.go:107] Error checking existing TLS certificate: secret "builder-tls" not found
I1128 10:01:07.298719       1 sync.go:209] Preparing certificate with issuer
I1128 10:01:07.299163       1 prepare.go:224] Compare "" with "https://acme-staging.api.letsencrypt.org/acme/reg/5145024"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x109da80]
 goroutine 229 [running]:
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).authorize(0xc420206280, 0x1cb35e0, 0xc420384800, 0xc42095e5f0, 0xc420582000, 0xc420964c40, 0x17, 0xc4202bdf10, 0x0, 0x0, ...)
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:137 +0x110
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).Prepare.func1(0xc4209639b0, 0xc420206280, 0x1cb35e0, 0xc420384800, 0xc42095e5f0, 0xc420582000, 0xc4207a73e0, 0xc420964c40, 0x17, 0xc4202bdf10, ...)
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:85 +0xce
created by github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).Prepare
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:91 +0x3ee
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.2", GitCommit:"922a86cfcd65915a9b2f69f3f193b8907d741d9c", GitTreeState:"clean", BuildDate:"2017-07-21T08:23:22Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.8-gke.0", GitCommit:"a7061d4b09b53ab4099e3b5ca3e80fb172e1b018", GitTreeState:"clean", BuildDate:"2017-10-10T18:48:45Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

```
- Cloud provider or hardware configuration**: GCE
- Install tools:
- Others:

Verbose logs:
```
I1128 11:38:15.327549       1 controller.go:111] Creating event broadcaster
I1128 11:38:15.329252       1 leaderelection.go:174] attempting to acquire leader lease...
I1128 11:38:15.329586       1 server.go:68] Listening on http://0.0.0.0:9402
I1128 11:38:15.342200       1 round_trippers.go:436] GET https://10.100.240.1:443/api/v1/namespaces/kube-system/endpoints/cert-manager-controller 200 OK in 12 milliseconds
I1128 11:38:15.348965       1 round_trippers.go:436] PUT https://10.100.240.1:443/api/v1/namespaces/kube-system/endpoints/cert-manager-controller 200 OK in 3 milliseconds
I1128 11:38:15.349758       1 leaderelection.go:184] successfully acquired lease kube-system/cert-manager-controller
I1128 11:38:15.350463       1 controller.go:63] Starting shared informer factory
I1128 11:38:15.350740       1 reflector.go:202] Starting reflector *v1beta1.Ingress (30s) from github.com/jetstack/cert-manager/third_party/k8s.io/client-go/informers/factory.go:72
I1128 11:38:15.350762       1 reflector.go:240] Listing and watching *v1beta1.Ingress from github.com/jetstack/cert-manager/third_party/k8s.io/client-go/informers/factory.go:72
I1128 11:38:15.350954       1 controller.go:113] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"cert-manager-controller", UID:"a30c83c8-d411-11e7-9b9c-42010a840fd1", APIVersion:"v1", ResourceVersion:"163809", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' cert-manager-4147707434-j0flw-external-cert-manager-controller became leader
I1128 11:38:15.351034       1 controller.go:54] Starting clusterissuers controller
I1128 11:38:15.351044       1 controller.go:98] Starting clusterissuers control loop
I1128 11:38:15.351081       1 controller.go:54] Starting certificates controller
I1128 11:38:15.351086       1 controller.go:146] Starting certificates control loop
I1128 11:38:15.351108       1 controller.go:54] Starting issuers controller
I1128 11:38:15.351117       1 controller.go:96] Starting issuers control loop
I1128 11:38:15.351374       1 reflector.go:202] Starting reflector *v1alpha1.ClusterIssuer (30s) from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I1128 11:38:15.351387       1 reflector.go:240] Listing and watching *v1alpha1.ClusterIssuer from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I1128 11:38:15.351803       1 reflector.go:202] Starting reflector *v1alpha1.Certificate (30s) from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I1128 11:38:15.351823       1 reflector.go:240] Listing and watching *v1alpha1.Certificate from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I1128 11:38:15.352127       1 reflector.go:202] Starting reflector *v1alpha1.Issuer (30s) from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I1128 11:38:15.352145       1 reflector.go:240] Listing and watching *v1alpha1.Issuer from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I1128 11:38:15.352445       1 reflector.go:202] Starting reflector *v1.Secret (30s) from github.com/jetstack/cert-manager/third_party/k8s.io/client-go/informers/factory.go:72
I1128 11:38:15.352468       1 reflector.go:240] Listing and watching *v1.Secret from github.com/jetstack/cert-manager/third_party/k8s.io/client-go/informers/factory.go:72
I1128 11:38:15.354807       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/extensions/v1beta1/ingresses?resourceVersion=0 200 OK in 3 milliseconds
I1128 11:38:15.362935       1 round_trippers.go:436] POST https://10.100.240.1:443/api/v1/namespaces/kube-system/events 201 Created in 12 milliseconds
I1128 11:38:15.363124       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/certmanager.k8s.io/v1alpha1/clusterissuers?resourceVersion=0 200 OK in 11 milliseconds
I1128 11:38:15.364524       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/certmanager.k8s.io/v1alpha1/issuers?resourceVersion=0 200 OK in 12 milliseconds
I1128 11:38:15.364783       1 round_trippers.go:436] GET https://10.100.240.1:443/api/v1/secrets?resourceVersion=0 200 OK in 12 milliseconds
I1128 11:38:15.365107       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/certmanager.k8s.io/v1alpha1/certificates?resourceVersion=0 200 OK in 13 milliseconds
I1128 11:38:15.367742       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/extensions/v1beta1/ingresses?resourceVersion=6283&timeoutSeconds=495&watch=true 200 OK in 6 milliseconds
I1128 11:38:15.374439       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/certmanager.k8s.io/v1alpha1/issuers?resourceVersion=155831&timeoutSeconds=554&watch=true 200 OK in 7 milliseconds
I1128 11:38:15.376260       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/certmanager.k8s.io/v1alpha1/clusterissuers?resourceVersion=162861&timeoutSeconds=356&watch=true 200 OK in 8 milliseconds
I1128 11:38:15.379547       1 round_trippers.go:436] GET https://10.100.240.1:443/apis/certmanager.k8s.io/v1alpha1/certificates?resourceVersion=162953&timeoutSeconds=452&watch=true 200 OK in 3 milliseconds
I1128 11:38:15.379652       1 round_trippers.go:436] GET https://10.100.240.1:443/api/v1/secrets?resourceVersion=155781&timeoutSeconds=361&watch=true 200 OK in 1 milliseconds
I1128 11:38:15.451216       1 shared_informer.go:116] caches populated
I1128 11:38:15.451265       1 shared_informer.go:116] caches populated
I1128 11:38:15.451294       1 controller.go:119] Starting "issuers" worker
I1128 11:38:15.451312       1 shared_informer.go:116] caches populated
I1128 11:38:15.451317       1 controller.go:152] Synced all caches for certificates control loop
I1128 11:38:15.451353       1 controller.go:121] Starting "clusterissuers" worker
I1128 11:38:15.451377       1 controller.go:138] clusterissuers controller: syncing item 'letsencrypt-staging'
I1128 11:38:15.451435       1 controller.go:170] Starting "certificates" worker
I1128 11:38:15.451449       1 controller.go:187] certificates controller: syncing item 'kube-system/builder-tls'
I1128 11:38:15.451508       1 acme.go:90] getting private key (letsencrypt-staging-issuer-account-key->tls.key) for acme issuer kube-system/letsencrypt-staging
I1128 11:38:15.451701       1 sync.go:107] Error checking existing TLS certificate: secret "builder-tls" not found
I1128 11:38:15.451885       1 sync.go:209] Preparing certificate with issuer
I1128 11:38:15.451910       1 acme.go:90] getting private key (letsencrypt-staging-issuer-account-key->tls.key) for acme issuer kube-system/letsencrypt-staging
I1128 11:38:15.452820       1 prepare.go:224] Compare "" with "https://acme-staging.api.letsencrypt.org/acme/reg/5145024"
I1128 11:38:15.453074       1 controller.go:119] Starting "issuers" worker
I1128 11:38:15.453091       1 controller.go:170] Starting "certificates" worker
I1128 11:38:15.453644       1 controller.go:113] Event(v1.ObjectReference{Kind:"Certificate", Namespace:"kube-system", Name:"builder-tls", UID:"a763ffcb-d42f-11e7-9b9c-42010a840fd1", APIVersion:"certmanager.k8s.io/v1alpha1", ResourceVersion:"162953", FieldPath:""}): type: 'Warning' reason: 'ErrorCheckCertificate' Error checking existing TLS certificate: secret "builder-tls" not found
I1128 11:38:15.453688       1 controller.go:113] Event(v1.ObjectReference{Kind:"Certificate", Namespace:"kube-system", Name:"builder-tls", UID:"a763ffcb-d42f-11e7-9b9c-42010a840fd1", APIVersion:"certmanager.k8s.io/v1alpha1", ResourceVersion:"162953", FieldPath:""}): type: 'Normal' reason: 'PrepareCertificate' Preparing certificate with issuer
I1128 11:38:15.454030       1 controller.go:121] Starting "clusterissuers" worker
I1128 11:38:15.458824       1 round_trippers.go:436] POST https://10.100.240.1:443/api/v1/namespaces/kube-system/events 201 Created in 5 milliseconds
I1128 11:38:15.463581       1 round_trippers.go:436] POST https://10.100.240.1:443/api/v1/namespaces/kube-system/events 201 Created in 4 milliseconds
I1128 11:38:15.947434       1 prepare.go:136] picking challenge type for domain "builder.int.company.com"
I1128 11:38:15.947564       1 sync.go:219] Issuing certificate...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x109da80]

goroutine 163 [running]:
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).authorize(0xc420244e80, 0x1cb35e0, 0xc420415600, 0xc42046b3b0, 0xc420222ea0, 0xc420444440, 0x17, 0xc420972150, 0x0, 0x0, ...)
        /go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:137 +0x110
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).Prepare.func1(0xc42080d0a0, 0xc420244e80, 0x1cb35e0, 0xc420415600, 0xc42046b3b0, 0xc420222ea0, 0xc4208617a0, 0xc420444440, 0x17, 0xc420972150, ...)
        /go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:85 +0xce
created by github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).Prepare
        /go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:91 +0x3ee

```

--- a/pkg/issuer/acme/prepare.go
+++ b/pkg/issuer/acme/prepare.go
@@ -23,6 +23,7 @@ const (
 	errorGetACMEAccount          = "ErrGetACMEAccount"
 	errorCheckAuthorization      = "ErrCheckAuthorization"
 	errorObtainAuthorization     = "ErrObtainAuthorization"
+	errorInvalidConfig           = "ErrInvalidConfig"
 
 	messageObtainedAuthorization    = "Obtained authorization for domain %s"
 	messagePresentChallenge         = "Presenting %s challenge for domain %s"
@@ -30,6 +31,7 @@ const (
 	messageErrorGetACMEAccount      = "Error getting ACME account: "
 	messageErrorCheckAuthorization  = "Error checking ACME domain validation: "
 	messageErrorObtainAuthorization = "Error obtaining ACME domain authorization: "
+	messageErrorMissingConfig       = "certificate.spec.acme must be specified"
 )
 
 // Prepare will ensure the issuer has been initialised and is ready to issue
@@ -38,6 +40,10 @@ const (
 // It will send the appropriate Letsencrypt authorizations, and complete
 // challenge requests if neccessary.
 func (a *Acme) Prepare(ctx context.Context, crt *v1alpha1.Certificate) error {
+	if crt.Spec.ACME == nil {
+		crt.UpdateStatusCondition(v1alpha1.CertificateConditionReady, v1alpha1.ConditionFalse, errorInvalidConfig, messageErrorMissingConfig)
+		return fmt.Errorf(messageErrorMissingConfig)
+	}
 	// obtain an ACME client
 	cl, err := a.acmeClient()
 	if err != nil {
@@ -133,6 +139,9 @@ func keyForChallenge(cl *acme.Client, challenge *acme.Challenge) (string, error)
 }
 
 func (a *Acme) authorize(ctx context.Context, cl *acme.Client, crt *v1alpha1.Certificate, auth authResponse) (*acme.Authorization, error) {
+	if crt.Spec.ACME == nil {
+		return nil, fmt.Errorf("certificate.spec.acme must be set")
+	}
 	glog.V(4).Infof("picking challenge type for domain %q", auth.domain)
 	challengeType, err := a.pickChallengeType(auth.domain, auth.auth, crt.Spec.ACME.Config)
 	if err != nil {

